### PR TITLE
Add support for editing import schedules

### PR DESCRIPTION
**Pull request by borkweb**
See: https://central.tri.be/issues/61161

### CODE REVIEW

**Comment by tr1b0t** [comment]
Status|Message|Line
:-----:|-----|-----
:warning: | Expected next thing to be an escaping function (see Codex for 'Data Validation'), not '$field' | [import-form.php:59](https://github.com/moderntribe/the-events-calendar/blob/801bdc565aeec002f2952f9a7d9e6d83675c35e5/src/admin-views/aggregator/tabs/import-form.php#L59)
:warning: | Expected next thing to be an escaping function (see Codex for 'Data Validation'), not '$field' | [import-form.php:78](https://github.com/moderntribe/the-events-calendar/blob/801bdc565aeec002f2952f9a7d9e6d83675c35e5/src/admin-views/aggregator/tabs/import-form.php#L78)

> via jenkins/codesniffer-phpcs

File: src/Tribe/Aggregator/Record/Abstract.php [hunk 1]
```diff
@@ -147,6 +148,72 @@ public function create( $type = 'manual', $args = array(), $meta = array() ) {
 
 		$meta = wp_parse_args( $meta, $defaults );
 
+		$post = $this->prep_post_args( $type, $args, $meta );
+
+		$result = wp_insert_post( $post );
+
+		// meta_input was introduced in 4.4. Deal with old versions
+		if ( -1 === version_compare( WP_VERSION, '4.4' ) && ! is_wp_error( $result ) ) {
+			foreach ( $post['meta_input'] as $key => $value ) {
```
Comment (bordoni) [inline src/Tribe/Aggregator/Record/Abstract.php]: No wonder, I didn't know about it... 😄

File: src/admin-views/aggregator/tabs/import-form.php [hunk 2]
```diff
@@ -0,0 +1,201 @@
+<?php
+$has_license_key = ! empty( Tribe__Events__Aggregator__Service::instance()->api()->key );
```
Comment (bordoni) [inline src/admin-views/aggregator/tabs/import-form.php]: There is a method inside of the Aggregator Main Class to check for that.

File: src/admin-views/aggregator/origins/meetup.php [hunk 3]
```diff
@@ -77,10 +77,15 @@
 			class="tribe-ea-field tribe-ea-dropdown tribe-ea-size-large"
 			placeholder="<?php echo esc_attr( $field->placeholder ); ?>"
 			data-hide-search
+			<?php if ( 'edit' === $aggregator_action ) : ?>
```
Comment (bordoni) [inline src/admin-views/aggregator/origins/meetup.php]: Could we make it a oneliner? (or it's too confusing?)

File: src/admin-views/aggregator/origins/ical.php [hunk 4]
```diff
@@ -25,10 +25,15 @@
 			class="tribe-ea-field tribe-ea-dropdown tribe-ea-size-large"
 			placeholder="<?php echo esc_attr( $field->placeholder ); ?>"
 			data-hide-search
+			<?php if ( 'edit' === $aggregator_action ) : ?>
```
Comment (bordoni) [inline src/admin-views/aggregator/origins/ical.php]: Same one-liner question

File: src/admin-views/aggregator/origins/meetup.php (hunk 3 above)
Comment (borkweb) [inline src/admin-views/aggregator/origins/meetup.php]: I prefer the 3-liner, personally. Accomplishes the same thing. :)

Comment (bordoni) [inline src/admin-views/aggregator/origins/meetup.php]: 😉

**Comment by bordoni** [comment]
👍

**Pull request merged**